iOS_SDK/OneSignalSDK/Source/OSStateSynchronizer.m

```diff
@@ -290,4 +291,22 @@ - (void)sendOnFocusTime:(NSNumber*)totalTimeActive
     }];
 }
 
+-(id)getResultByChannel:(NSDictionary<NSString *, id> *)results {
```

Comment:
Why have two lists (`OS_SECONDARY_CHANNELS` & `OS_SECONDARY_CHANNELS `) instead of one `OS_CHANNELS` as ` @[OS_PUSH, OS_EMAIL]`?

Then this code could be:
```objective-c
   for (NSString* channelId in OS_CHANNELS) {
        result = results[channelId];
        if (result)
            return result;
    }
   return nil;
```

Reply:
Also this get the first result and the name doesn't reflect that. A better name might be `getFirstAvailableResult` or `getFirstResultByChannelPriority`

Reply:
> Why have two lists (`OS_SECONDARY_CHANNELS` & `OS_SECONDARY_CHANNELS `) instead of one `OS_CHANNELS` as ` @[OS_PUSH, OS_EMAIL]`?
> 
> Then this code could be:
> 
> ```objc
>    for (NSString* channelId in OS_CHANNELS) {
>         result = results[channelId];
>         if (result)
>             return result;
>     }
>    return nil;
> ```

You are right, I was thinking of having different types of priority channels, but this can be simplified. At this level, there is no need for this abstraction